azuredevops/internal/service/memberentitlementmanagement/resource_service_principal_entitlement.go

```diff
@@ -0,0 +1,322 @@
+package memberentitlementmanagement
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ahmetb/go-linq"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/graph"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/licensing"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/memberentitlementmanagement"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/webapi"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/suppress"
+)
+
+func ResourceServicePrincipalEntitlement() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServicePrincipalEntitlementCreate,
+		Read:   resourceServicePrincipalEntitlementRead,
+		Update: resourceServicePrincipalEntitlementUpdate,
+		Delete: resourceServicePrincipalEntitlementDelete,
+		Importer: &schema.ResourceImporter{
+			State: importServicePrincipalEntitlement,
+		},
+		Schema: map[string]*schema.Schema{
+			"origin_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+			"origin": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      string("aad"),
```

Comment:
any other possible values ​​besides the default value?
```suggestion
				Default:      "aad",
```